src/Log.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Log.h is part of Brewtarget, and is Copyright the following
+ * authors 2009-2015
+ * - Maxime Lavigne <duguigne@gmail.com>
+ *
+ * Brewtarget is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Brewtarget is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _LOG_H
+#define _LOG_H
+
+#include <QFile>
+#include <QObject>
+#include <QMutex>
+#include <QString>
+#include <QTextStream>
+
+/*!
+ * \class Log
+ *
+ * \brief Provides a proxy to an OS agnostic Log file.
+ */
+class Log : public QObject
+{
+   Q_OBJECT
+
+public:
+   Log(const QString defaultDir, bool isLoggingToStderr = true);
```

Comment:
Is this class meant to be instantiated only in `class Brewtarget`? If so, we might want to make the constructor private, and add `Brewtarget` as a friend class to make it clear that you shouldn't create instances by yourself.